kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/PodOperationsImpl.java

```diff
@@ -256,6 +258,13 @@ public ExecWatch exec(String... command) {
             } else {
                 sb.append("&command=");
             }
+          
+            try {
```

Comment:
Can you use try-with-resources here ? It would be better
```
try ( cmd = URLEncoder.encode(cmd, "UTF-8") ) {
   sb.append(cmd);
} catch (ex........) { ... }
```

Also, You can create a URL utility method to wrap  ```URLEncoder.encode(cmd, "UTF-8")``` , like @rohanKanojia suggested.

Reply:
@dev-gaur 
URLEncoder.encode(cmd, "UTF-8") returns a String. This cannot be used in Try-With-Resources as it expects an instance that implements AutoCloseable.
Please let me know if my understanding is different.